pkg/controller/plan/adapter/vsphere/builder.go

```diff
@@ -226,6 +227,20 @@ func (r *Builder) PodEnvironment(vmRef ref.Ref, sourceSecret *core.Secret) (env
 			Value: macsToIps,
 		})
 	}
+
+	if libref.RefSet(&vm.LUKS) {
```

Comment:
`libref.RefSet` checks both name and namespace. when adding envVar from secret we can set only the name.
@ahadas do you think we need to change this to check only the name isn't empty(both here an API)?